api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/ContrastAccountServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContrastAccountServiceImpl.class);
+    private static final String CONTRAST = "contrast";
+    private static final String VALIDATION_FAILED = "Error in validating account";
+    private static final String CONTRAST_TOKEN_TEMPLATE = "{\"api_key\":\"%s\"," +
+            "\"authorization\":\"%s\",\"env_name\":\"%s\"}";
+    private static final String CONTRAST_URL_TEMPLATE = "https://%s.contrastsecurity.com";
+    private static final String ORGANIZATIONS_URL_PATH = "/api/v4/organizations/";
+    private static final String CONTRAST_ENABLED = CONTRAST + ".enabled";
+
+    private final CredentialProvider credentialProvider;
+    private final ObjectMapper objectMapper;
+
+    public ContrastAccountServiceImpl(CredentialProvider credentialProvider, ObjectMapper objectMapper) {
+        this.credentialProvider = credentialProvider;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public String serviceType() {
+        return CONTRAST;
+    }
+
+    @Override
+    public AccountValidationResponse validate(CreateAccountRequest accountData) {
+        AccountValidationResponse validationResponse = validateRequestData(accountData);
+        if (validationResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info("Validation failed due to missing parameters");
+            return validationResponse;
+        }
+        return getOrganizationName(accountData);
+    }
+
+    private AccountValidationResponse getOrganizationName(CreateAccountRequest pluginData) {
+        AccountValidationResponse validationResponse = new AccountValidationResponse();
+        validationResponse.setValidationStatus(FAILURE);
+        validationResponse.setMessage(VALIDATION_FAILED);
+        try {
+            String apiUrl = new URIBuilder(String.format(CONTRAST_URL_TEMPLATE, pluginData.getContrastEnvironmentName()))
+                    .setPath(ORGANIZATIONS_URL_PATH + pluginData.getContrastOrganizationId()).build().toString();
+            String authKey = Base64.getEncoder().encodeToString((pluginData.getContrastUserId() + ":"
+                    + pluginData.getContrastServiceKey()).getBytes());
+            HttpGet request = new HttpGet(apiUrl);
+            request.setHeader(HttpHeaders.AUTHORIZATION, authKey);
+            request.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+            request.setHeader("API-Key", pluginData.getContrastApiKey());
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                try (CloseableHttpResponse response = httpClient.execute(request)) {
+                    int statusCode = response.getStatusLine().getStatusCode();
+                    if (statusCode != 200) {
+                        String errorMessage = handleErrorResponse(response.getStatusLine().getStatusCode());
+                        validationResponse.setErrorDetails(errorMessage);
```

Comment:
Set it as a message as well

Reply:
not needed since message is already set as "Contrast validation Failed"